cypress/e2e/shared/fluxQueryBuilder.test.ts

```diff
@@ -218,33 +249,37 @@ describe('Script Builder', () => {
         }
       }
 
+      beforeEach(() => {
+        cy.log('empty downloads directory')
+        cy.task('deleteDownloads', {dirPath: downloadsDirectory})
+      })
+
       it('will return 0 tables and 0 rows, for an empty dataset', () => {
+        cy.log('turn off composition sync')
+        cy.getByTestID('flux-sync--toggle').click()
+        cy.getByTestID('flux-sync--toggle').should('not.have.class', 'active')
         cy.log('select empty dataset')
-        selectBucket('defbuck3')
-        cy.getByTestID('flux-editor').contains(`from(bucket: "defbuck3")`, {
-          timeout: 3000,
-        })
+        cy.getByTestID('flux-editor').monacoType(`{selectall}{enter}
+            from(bucket: "defbuck3") |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
+          `)
+        cy.getByTestID('flux-editor').contains('defbuck3')
 
         runTest(0, 0, false)
       })
 
-      describe('will return 1 table', () => {
-        beforeEach(() => {
-          const writeData: string[] = []
-          writeData.push(`ndbc3,air_temp_degc=70_degrees station_id=${i}`)
-          cy.writeData(writeData, 'defbuck4')
-          cy.wait(1200)
+      it('will return 1 table, for a dataset with only 1 table', () => {
+        cy.log('select dataset with 1 table')
+        selectBucket('defbuck4')
+        cy.getByTestID('flux-editor').contains(`from(bucket: "defbuck4")`, {
+          timeout: 3000,
```

Comment:
We use timeouts associated with the flux-editor for two reasons:
1. Because the flux editor is lazy loaded. So the react component is rendered after everything else on the page.
2. Then the LSP server (in it's own thread) takes time to come online. So any composition commands (such as selecting the bucket) will not result in changed code block,  until the LSP is fully booted up.

I would also note that these are optional waits. Cypress does not have to wait for the entire duration before checking...it's just a "wait until" with a longer timeout before quitting.